packages/core-data/src/actions.js

```diff
@@ -132,14 +215,33 @@ export function* saveEntityRecord( kind, name, record ) {
 	}
 	const key = entity.key || DEFAULT_ENTITY_KEY;
 	const recordId = record[ key ];
-	const updatedRecord = yield apiFetch( {
-		path: `${ entity.baseURL }${ recordId ? '/' + recordId : '' }`,
-		method: recordId ? 'PUT' : 'POST',
-		data: record,
-	} );
-	yield receiveEntityRecords( kind, name, updatedRecord, undefined, true );
-
-	return updatedRecord;
+
+	yield { type: 'SAVE_ENTITY_RECORD_START', kind, name, recordId };
+	let error;
+	try {
+		const updatedRecord = yield apiFetch( {
+			path: `${ entity.baseURL }${ recordId ? '/' + recordId : '' }`,
+			method: recordId ? 'PUT' : 'POST',
+			data: record,
+		} );
+		yield receiveEntityRecords( kind, name, updatedRecord, undefined, true );
+	} catch ( _error ) {
+		error = _error;
+	}
+	yield { type: 'SAVE_ENTITY_RECORD_FINISH', kind, name, recordId, error };
+}
+
+/**
+ * Action triggered to save an entity record's edits.
+ *
+ * @param {string} kind     Kind of the entity.
+ * @param {string} name     Name of the entity.
+ * @param {Object} recordId ID of the record.
+ */
+export function* saveEditedEntityRecord( kind, name, recordId ) {
+	const edits = yield select( 'getEntityRecordEdits', kind, name, recordId );
```

Comment:
should we consider returning early if there's no edits?

Reply:
Yes